src/Microsoft.AspNetCore.Server.Kestrel/Infrastructure/MemoryPoolIterator.cs

```diff
@@ -663,6 +663,109 @@ public int GetLength(MemoryPoolIterator end)
             }
         }
 
+        public static unsafe void FastByteCopy(byte[] srcArray, uint srcOffset, byte[] destArray, uint destOffset, uint byteCount)
+        {
+            Debug.Assert(srcOffset + byteCount <= srcArray.Length);
+            Debug.Assert(destOffset + byteCount <= destArray.Length);
+            
+            if (byteCount < 64)
+            {
+                // Buffer.BlockCopy is expensive for small copies.  Do it manually ourselves.
+                
+                fixed (byte* pSrc = srcArray)
+                fixed (byte* pDest = destArray)
```

Comment:
Oddly, I get an IndexOutOfRange exception during unit tests with "byte* pDest = &destArray[0]", whereas I don't with "byte* pDest = destArray".

Are you sure the info from #541 is correct?